lib/puppet/util/windows/file.rb

```diff
@@ -179,12 +200,11 @@ def self.device_io_control(handle, io_control_code, in_buffer = nil, out_buffer
 
   FILE_ATTRIBUTE_REPARSE_POINT = 0x400
   def symlink?(file_name)
-    begin
-      attributes = get_attributes(file_name)
-      (attributes & FILE_ATTRIBUTE_REPARSE_POINT) == FILE_ATTRIBUTE_REPARSE_POINT
-    rescue
-      # raised INVALID_FILE_ATTRIBUTES is equivalent to file not found
+    attributes = get_attributes(file_name, false)
```

Comment:
It's too bad that `INVALID_FILE_ATTRIBUTES` is all 1s.... but you can still remove the extra `else` branch if you want:

```ruby
def symlink?(file_name)
  attributes = get_attributes(file_name, false)

  return false if (attributes == INVALID_FILE_ATTRIBUTES)
  (attributes & FILE_ATTRIBUTE_REPARSE_POINT) == FILE_ATTRIBUTE_REPARSE_POINT
end
```